static/src/javascripts/projects/commercial/modules/creatives/page-skin.js

```diff
@@ -40,19 +41,66 @@ const pageSkin = (): void => {
         }
     };
 
-    const initTopPositionOnce = (): void => {
+    const initTopPositionOnce = (hasTruskin: boolean): void => {
         if (topPosition === 0) {
             const navHeader = document.getElementsByClassName('new-header')[0];
             if (navHeader) {
-                topPosition = navHeader.offsetTop + navHeader.offsetHeight;
+                topPosition = hasTruskin
+                    ? navHeader.offsetTop + adLabelHeight
+                    : navHeader.offsetTop + navHeader.offsetHeight;
             }
         }
     };
 
-    // This is to reposition the Page Skin to start where the navigation header ends.
+    const shrinkElement = (element: HTMLElement | null): void => {
+        const frontContainer = document.querySelector('.fc-container__inner');
+        if (element && frontContainer) {
+            element.style.cssText = `max-width: ${
+                frontContainer.clientWidth
+            }px; margin-right: auto; margin-left: auto;`;
+        }
+    };
+
     const repositionSkin = (): void => {
-        if (hasPageSkin && isInAUEdition) {
-            initTopPositionOnce();
+        const bodyElement = document.querySelector('body');
+        const hasTruskin = bodyElement
+            ? bodyElement.classList.contains('truskin-page-skin')
+            : false;
+        const header = document.querySelector('.new-header');
+        const footer = document.querySelector('.l-footer');
+        const topBannerAd = document.querySelector('.ad-slot--top-banner-ad');
+
+        if (hasTruskin && header && topBannerAd) {
+            initTopPositionOnce(hasTruskin);
+
+            shrinkElement(header);
+            shrinkElement(footer);
```

Comment:
We could wrap `shrinkElement(footer)` in a conditional `if (footer) { shrinkElement(footer) }` check? 

Because we're not checking for the existence of `footer` on line 73 https://github.com/guardian/frontend/pull/22283/files#diff-c65ce2ed71ff68922460abf7bcd11cccR73 we could be passing `null` to `shrinkElement`.

I assume this is why the type for the `element` parameter on the `shrinkElement` function is `HTMLElement | null`, this could be simplified to `HTMLElement` if we use the `if (footer) { shrinkElement(footer) }` check.

Reply:
Good point, we just added the existence checking inside the function itself:
```javascript
if (element && frontContainer) {
    element.style.cssText = `max-width: ${
        frontContainer.clientWidth
    }px; margin-right: auto; margin-left: auto;`;
}
```

 We now instead changed it as you suggested for better readability